client/src/app/site/tags/models/view-tag.ts

```diff
@@ -0,0 +1,51 @@
+import { Tag } from '../../../shared/models/core/tag';
+import { TranslateService } from '@ngx-translate/core';
+import { BaseViewModel } from '../../base/base-view-model';
+
+/**
+ * Category class for the View
+ *
+ * Stores a Category including all (implicit) references
+ * Provides "safe" access to variables and functions in {@link Category}
+ * @ignore
+ */
+export class ViewTag extends BaseViewModel {
+    private _tag: Tag;
+
+    public constructor(tag: Tag) {
+        super();
+        this._tag = tag;
+    }
+
+    public get tag(): Tag {
+        return this._tag;
+    }
+
+    public set tag(tag: Tag) {
+        this._tag = tag;
+    }
+
+    public get id(): number {
+        return this.tag ? this.tag.id : null;
+    }
+
+    public get name(): string {
+        return this.tag ? this.tag.name : null;
+    }
+
+    public set name(name: string) {
+        this._tag.name = name;
+    }
+
+    public getTitle(translate?: TranslateService): string {
```

Comment:
having the translate service in the getTitle function is (or will be) deprecated.
Might change after a rebase